packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs

```diff
@@ -24,12 +24,13 @@ public async Task ExecuteAsync()
         {
             GeneratedCodeWorkspace.Initialize();
             var outputPath = CodeModelPlugin.Instance.Configuration.OutputDirectory;
-            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", "Generated");
+            var generatedSourceOutputPath = ParseGeneratedSourceOutputPath(outputPath);
+            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", GeneratedFolderName);
 
             GeneratedCodeWorkspace workspace = await GeneratedCodeWorkspace.Create();
 
             var output = CodeModelPlugin.Instance.OutputLibrary;
-            Directory.CreateDirectory(Path.Combine(outputPath, "src", "Generated", "Models"));
+            Directory.CreateDirectory(Path.Combine(generatedSourceOutputPath, "Models"));
```

Comment:
If we are creating the directory here https://github.com/microsoft/typespec/pull/3616/files#diff-9bfb4971a041afb747e174ac1e942fb8e1fe1343bee198af96931f0df82fdb2bR85, why is this line needed?